waspc/src/Wasp/Generator/ServerGenerator/Auth/OAuthAuthG.hs

```diff
@@ -98,18 +99,16 @@ genOAuthProvider provider maybeUserConfig
 genOAuthConfig ::
   OAuthAuthProvider ->
   Maybe AS.Auth.ExternalAuthConfig ->
-  Path' (Rel ServerSrcDir) File' ->
+  Path' (Rel ServerTemplatesSrcDir) File' ->
   Generator FileDraft
-genOAuthConfig provider maybeUserConfig pathToConfigDst = return $ C.mkTmplFdWithDstAndData tmplFile dstFile (Just tmplData)
+genOAuthConfig provider maybeUserConfig pathToConfigTmpl = return $ C.mkTmplFdWithData tmplFile (Just tmplData)
```

Comment:
We used `_oauth.ts` as a common template before, but now each provider has their own template file because their code now differs (the way we fetch the user info, some providers require extra steps, some not)

Reply:
I've given it a shot to minimise the repeated code. We could probably go further, but I didn't to be too smart about inventing new abstractions, this now feels okay, check it out and tell me what do you think. Should we go further?

Reply:
I think that's enough abstraction for now. We'll come back to it when we add more providers and are smarter.